### PR TITLE
Added tests for @SqlInsert, @SqlUpdate, and @SqlDelete containing stored procedures

### PR DESCRIPTION
**Pull request by blafond**
Fixes [#592](https://github.com/hibernate/hibernate-reactive/issues/592)

### CODE REVIEW

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/CustomStoredProcecureSqlTest.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import io.vertx.ext.unit.TestContext;
+
+import static org.hibernate.reactive.containers.DatabaseConfiguration.DBType.POSTGRESQL;
+import static org.hibernate.reactive.util.impl.CompletionStages.completedFuture;
+
+public class CustomStoredProcecureSqlTest extends BaseReactiveTest {
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/CustomStoredProcecureSqlTest.java:36]: CustomStoredProce**d**ureSqlTest

**Review by DavideD — COMMENTED** [review]
Just need to rename the class name.

Btw, did you check if stores procedures work for MySQL and Db2?

I'm asking because sometimes the drivers don't recognize the SQL.
Like for this issue for example: #560

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/CustomStoredProcecureSqlTest.java (hunk 1 above)
Comment (blafond) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/CustomStoredProcecureSqlTest.java:36]: geesh.. been staring at the class for a few days.. should have run spell check I guess

Comment (blafond) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/CustomStoredProcecureSqlTest.java:36]: Have not checked mysql or db2 but will.

**Comment by gavinking** [comment]
Excellent, thanks for this @blafond!

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/CustomStoredProcecureSqlTest.java (hunk 1 above)
Comment (DavideD) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/CustomStoredProcecureSqlTest.java:36]: > Have not checked mysql or db2 but will.

We can probably create a separate test for this

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/CustomStoredProcecureSqlTest.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+							context.assertNotNull(foundRecord.updated);
+							context.assertNotNull(foundRecord.deleted);
+						} )
+		);
+	}
+
+	@NamedStoredProcedureQueries(
+			value = {
+					@NamedStoredProcedureQuery(
+							name = "SimpleRecord.insertRecord",
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/CustomStoredProcecureSqlTest.java:151]: Why did you use these annotations?

**Comment by DavideD** [comment]
Superseeded by #635

**Pull request closed without merging**